### PR TITLE
put more punctuation characters in the never-identifier category

### LINKED ISSUE #30327: put bracket characters in the non-identifier category of characters
ERROR: LoadError: LoadError: syntax: interpolated variable $j ends with invalid character "⟩"; use "$(j)" instead.
Example: "START smoothing ⟨$i|∂/∂R|$j⟩"
It worked in Julia 0.5

### PR DESCRIPTION
**Pull request by JeffBezanson**
fixes #30327

### CODE REVIEW

File: src/flisp/julia_extensions.c [hunk 1]
```diff
@@ -171,7 +171,18 @@ static int never_id_char(uint32_t wc)
...
+          wc == '`' ||
+
+          // mathematical brackets
+          (wc >= 0x27e6 && wc <= 0x27ef) ||
+          // angle, corner, and lenticular brackets
+          (wc >= 0x3008 && wc <= 0x3011) ||
+          // tortoise shell, square, and more lenticular brackets
+          (wc >= 0x3014 && wc <= 0x301b) ||
+          // some fullwidth punctuation
+          (wc >= 0xff08 && wc <= 0xff0f) ||
```
Comment (stevengj) [inline src/flisp/julia_extensions.c]: Are we 100% sure that something like `＋` (0xff0b)` will never be a valid identifier like `+`?  I'd suggest leaving out the non-bracket fullwidth chars to be conservative.

**Comment by stevengj** [comment]
Maybe it would be better to just exclude anything in categories Ps or Pe (punctuation open and close), which includes all of these brackets?

File: src/flisp/julia_extensions.c (hunk 1 above)
Comment (JeffBezanson) [reply src/flisp/julia_extensions.c]: Ah, right. I forgot that we `or` `opchar?` and `never-identifier-char?` separately.

**Comment by JeffBezanson** [comment]
I thought of that, but that also includes superscript and subscript brackets, which are allowed in identifiers, and also some ambiguous characters whose possible uses are too uncertain.

**Review by stevengj — APPROVED** [review]

**Pull request merged**